### PR TITLE
NH-3988 - Replace ApplicationException with Exception

### PR DESCRIPTION
**Pull request by hazzik**

### CODE REVIEW

File: src/NHibernate/Logging.cs [hunk 1]
```diff
@@ -63,15 +63,15 @@ private static ILoggerFactory GetLoggerFactory(string nhibernateLoggerClass)
 			}
 			catch (MissingMethodException ex)
 			{
-				throw new ApplicationException("Public constructor was not found for " + loggerFactoryType, ex);
+				throw new Exception("Public constructor was not found for " + loggerFactoryType, ex);
```
Comment (fredericDelaporte) [inline src/NHibernate/Logging.cs]: I would rather throw an ArgumentException here (or InvalidOperationException). We should not throw the base Exception class.

File: src/NHibernate/Logging.cs [hunk 2]
```diff
@@ -63,15 +63,15 @@ private static ILoggerFactory GetLoggerFactory(string nhibernateLoggerClass)
...
 			}
 			catch (InvalidCastException ex)
 			{
-				throw new ApplicationException(loggerFactoryType + "Type does not implement " + typeof (ILoggerFactory), ex);
+				throw new Exception(loggerFactoryType + "Type does not implement " + typeof (ILoggerFactory), ex);
```
Comment (fredericDelaporte) [inline src/NHibernate/Logging.cs]: Same as previous.

File: src/NHibernate/Logging.cs [hunk 3]
```diff
@@ -63,15 +63,15 @@ private static ILoggerFactory GetLoggerFactory(string nhibernateLoggerClass)
...
 			}
 			catch (Exception ex)
 			{
-				throw new ApplicationException("Unable to instantiate: " + loggerFactoryType, ex);
+				throw new Exception("Unable to instantiate: " + loggerFactoryType, ex);
```
Comment (fredericDelaporte) [inline src/NHibernate/Logging.cs]: Same as previous.

**Comment by hazzik** [comment]
Replaced bare `Exception` with `InstantiationException`

**Pull request merged**